.github/workflows/general.yaml

```diff
@@ -0,0 +1,59 @@
+name: "general"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  cargo_test:
+    name: cargo_test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.84.0
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo test --no-fail-fast
+
+  rustfmt:
```

Comment:
The time taken to install Rust and rebuild dependencies is going to outweigh the speed gain of running these as separate steps. I suggest that the workflow run these checks as steps rather than jobs